Adding more context around link usage in buttons (#1629)
hashicorp/design-system: website/docs/components/button/partials/guidelines/guidelines.md

### When to use

- To trigger an action or event, like a form submission.

> - For links that require more prominence, such as CTAs (calls-to-action), see [Links in Buttons](/components/button#links-in-buttons).

Reviewer: ```suggestion
- For links that require more prominence, such as CTAs (calls-to-action), see [Links as Buttons](/components/button#links-as-buttons).
```

> - For links that are part of an [inverse button group](/patterns/button-organization#grouping), see [Links in Buttons](/components/button#links-in-buttons).

Reviewer: ```suggestion
- For links that are part of an [inverse button group](/patterns/button-organization#grouping), see [Links as Buttons](/components/button#links-as-buttons).
```

<Hds::Button @color="secondary" @text="Authenticate with GitHub" @icon="external-link" @iconPosition="trailing" @href="https://hashicorp.com" />

> ## Links in Buttons

Reviewer: ```suggestion
## Links as Buttons
```

!!! Insight

> While we support using links within buttons, please strive for the correct semantic use. Learn more about [how semantic use helps make our products more conformant](/components/button?tab=accessibility#button-vs-link).

Reviewer: ```suggestion
While we support using links as buttons (that is, with the visual appearance of a button), please strive for the correct semantic use. Learn more about [how semantic use helps make our products more conformant](/components/button?tab=accessibility#button-vs-link).
```

Author: I ended up taking this out since it was essentially sending the same message as the content below but in different words.

!!!

> The Button component accepts an `@href` or `@route` argument, which results in a link with the visual appearance of a button. While we advise against this approach because it can cause confusion for keyboard-only users, there are a few instances where this may be appropriate for improved visual hierarchy, such as:

Reviewer: ```suggestion
The Button component accepts an `@href` or `@route` argument, which results in a link with the visual appearance of a button. While in general we advise against this approach because it can cause confusion for keyboard-only users, there are a few instances where this may be appropriate for improved visual hierarchy, such as:
```

Reviewer: Another consideration: above we say "While we support using links as buttons" and here we say "While we advise against this approach". Could it be seen as contradictory?

For links that require more prominence and CTAs (calls-to-action).

<Hds::Button @color="primary" @text="Sign up for free" @size="large" @href="#" />

<Hds::Alert @type="inline" @color="neutral" as |A|>
  <A.Title>Modifying user permissions</A.Title>
  <A.Description>

>     This page displays all users in the organization that have permissions in this project. A user must be invited to the org first before you can change the user’s role at the project level.

Reviewer 2: ```suggestion
    This page displays all users in the organization who have permissions in this project. A user must be invited to the org first before you can change the user’s role at the project level.
```
